shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1048,7 +1048,7 @@ - (BOOL)canBecomeFirstResponder {
 - (BOOL)canPerformAction:(SEL)action withSender:(id)sender {
   // When scribble is available, the FlutterTextInputView will display the native toolbar unless
   // these text editing actions are disabled.
-  if (IsScribbleAvailable()) {
+  if ([self isScribbleAvailable] && sender == NULL) {
```

Comment:
@justinmc do you remember what does `sender == NULL` mean here? 

Also can you explain more about what this is trying to do? When I remove this `sender == NULL`, CMD+C/V from hardware keyboard still works. So I may be missing somehting. Thanks!

Reply:
Sorry I didn't document this better, I'm struggling to remember...

Are you trying this on an iPad with a hardware keyboard attached?  And iOS 14+.

The relevant part of the [docs for canPerformAction](https://developer.apple.com/documentation/uikit/uiresponder/1621105-canperformaction) say:

 > The object calling this method. For the editing menu commands, this is the shared [UIApplication](https://developer.apple.com/documentation/uikit/uiapplication) object.

Looking at the original scribble PR (https://github.com/flutter/engine/pull/24224), we seem to call `performAction` a bunch.  Maybe it's called with no `sender` from some of those and/or when hardware keys are being pressed?

Are you considering removing that expression for some reason?  I would test a physical iPad and Apple Pencil and hardware keyboard before at least.

Reply:
I am not doing anything particular for this code. I was just looking into "secure paste" feature, and trying to understand the codebase.

Reply:
interestingly i found that this function is not even called when we copy/paste. The other function in `FlutterPlatformPlugin.mm` is called instead. This code seems to be obsolete

Reply:
Nice, I'm happy to review if you want to remove it.